source/core/TypeAndDataManager.cpp

```diff
@@ -922,6 +924,20 @@ Boolean TypeCommon::IsComplex()
     }
     return false;
 }
+
+//------------------------------------------------------------
+Boolean TypeCommon::IsNamedClusterDataType() {
```

Comment:
Please rename this to something like IsIntrinsicClusterDataType().
Via-code-defined clusters can have names, but that's not what this is checking.